src/createTimestampRecord.ts

```diff
@@ -1,54 +1,55 @@
 import logger from './util/logger';
 import _ from 'lodash';
-import { LegacyTechRecord, SingleTechRecord } from './Interfaces/ILegacyTechRecord';
-import { NewKeyStructure } from './Interfaces/INewTechRecord';
+import { LegacyVehicleRecord, LegacyTechnicalRecord } from './interfaces/LegacyVehicleRecord';
+import { NewVehicleRecord } from './interfaces/NewVehicleRecord';
+import { PrimitiveTypes } from './interfaces/PrimitiveTypes';
 
-const isValidValue = (a: unknown) => {
-  return a !== null && a !== undefined && (_.isString(a) || _.isNumber(a) || _.isBoolean(a));
+const isDefined = (a: unknown) => {
+  return typeof a !== 'undefined';
 };
 
-const flattenAttributes = (vehicle: NewKeyStructure, recordPiece: object, prefix: string) => {
-  if (recordPiece === null || recordPiece === undefined) {
+const flattenAttributes = (vehicle: NewVehicleRecord, recordPiece: object, prefix: string) => {
+  if (!isDefined(recordPiece)) {
     return;
   }
```

Comment:
```suggestion
const flattenAttributes = (vehicle: NewVehicleRecord, recordPiece: object | undefined, prefix: string) => {
  if (!isDefined(recordPiece)) {
    return;
  }
```
Just to keep things consistent. Because the type of `recordPiece` means it should always be defined, so the `if` would be superfluous